File: src/tafqeetj/numbers/ThreeDigitsNumber.java
package tafqeetj.numbers;

import java.util.Objects;

public class ThreeDigitsNumber {
    private final int number;

    private ThreeDigitsNumber(int number) {
        if (number < 0 || number > 999)
            throw new IllegalArgumentException("Number must be between 1 and 999 inclusive "+number);
        this.number = number;
    }

    public static ThreeDigitsNumber of(int number) {
        return new ThreeDigitsNumber(number);
    }

    public int getNumber() {
        return number;
    }

    public boolean containsHundreds() {
        return number / 100 != 0;
    }

    public boolean containsTens() {
        return (number % 100) / 10 != 0;
    }

    public boolean containsOnes() {
        return (number % 100) % 10 != 0;
    }

    public boolean containsComposedNumber() {
        return containsTens() && containsOnes() && (number % 100) / 10 == 1;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ThreeDigitsNumber that = (ThreeDigitsNumber) o;
        return number == that.number;
    }

    @Override
    public int hashCode() {
        return Objects.hash(number);
    }

    @Override
    public String toString() {
        return "ThreeDigitsNumber{" +
                "number=" + number +
                '}';
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/numbers/ThreeDigitsNumber.kt
package tech.moussa.tafqeetk.numbers

import java.util.Objects

class ThreeDigitsNumber(
    private val number: Int = 0,
    private val sign: NumberSign = NumberSign.POSITIVE
) {
    init {
        require(!(this.number < 0 || this.number > 999)) { "Number must be between 0 and 999 inclusive: $number" }
    }

    fun getNumber(): Int {
        return this.number
    }

    fun getSign(): NumberSign {
        return this.sign
    }

    fun containsHundreds(): Boolean {
        return this.number / 100 != 0
    }

    fun containsTens(): Boolean {
        return this.number % 100 / 10 != 0
    }

    fun containsOnes(): Boolean {
        return this.number % 100 % 10 != 0
    }

    fun containsComposedNumber(): Boolean {
        return containsTens() && containsOnes() && number % 100 / 10 == 1
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as ThreeDigitsNumber
        return this.number == that.number
    }

    override fun hashCode(): Int {
        return Objects.hash(this.number)
    }

    override fun toString(): String {
        return "ThreeDigitsNumber{" +
                "number=" + this.number +
                '}'
    }

    companion object {
        fun of(number: Int, sign: NumberSign): ThreeDigitsNumber {
            return ThreeDigitsNumber(number, sign)
        }
    }
}